Search doesn't work properly

Pre-conditions:
Env. - Solus linux (unstable repository) https://github.com/getsolus/packages/pull/6573
Installed software e.g. ghostty

Description:
Testing hyprlauncher 0.1.1 on Solus linux

Reproduction steps:

1. Open hyprlauncher
2. Write ghostty in search field

Expected results:
Ghostty terminal should be displayed on first place

Currently:

1. Ghostty terminal is absent in hyprlauncher
2. Navigating by keyboard -> arrow down doesn't scroll 

<img width="1465" height="916" alt="Image" src="https://github.com/user-attachments/assets/5d81290f-da4a-46b1-8e5c-e5bf224fe25c" />

Latest git fails for me to build:
`FAILED: CMakeFiles/hyprlauncher.dir/src/main.cpp.o
/usr/lib64/ccache/bin/x86_64-solus-linux-g++ -DHYPRLAUNCHER_VERSION=\"0.1.0\" -I/home/build/YPKG/root/hyprlauncher/build/hyprlauncher-main/./protocols -isystem /usr/include/pixman-1 -isystem /usr/include/libdrm -isystem /usr/include/libxml2 -mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -fno-plt -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wall -Wno-error -Wp,-D_REENTRANT -O2 -g -DNDEBUG -std=gnu++23 -Wall -Wextra -Wno-unused-parameter -Wno-unused-value -Wno-missing-field-initializers -Wpedantic -O3 -DWITH_GZFILEOP -MD -MT CMakeFiles/hyprlauncher.dir/src/main.cpp.o -MF CMakeFiles/hyprlauncher.dir/src/main.cpp.o.d -o CMakeFiles/hyprlauncher.dir/src/main.cpp.o -c /home/build/YPKG/root/hyprlauncher/build/hyprlauncher-main/src/main.cpp
In file included from ../src/socket/ClientSocket.hpp:4,
                 from ../src/main.cpp:7:
../protocols/hyprlauncher_core-client.hpp:57:31: error: \u2018options\u2019 has not been declared
   57 |     void sendOpenWithOptions( options);
      |                               ^~~~~~~
[13/21] Building CXX object CMakeFiles/hyprlauncher.dir/src/query/QueryProcessor.cpp.o
ninja: build stopped: subcommand failed.
Command '['/bin/bash', '--norc', '--noprofile', '/tmp/ypkg-buildt96mpjne']' returned non-zero exit status 1.`

--- a/src/finders/desktop/DesktopFinder.cpp
+++ b/src/finders/desktop/DesktopFinder.cpp
@@ -80,10 +80,7 @@ class CDesktopEntry : public IFinderResult {
     uint32_t    m_frequency = 0;
 };
 
-static constexpr std::array<const char*, 3> DESKTOP_ENTRY_PATHS = {"/usr/local/share/applications", "/usr/share/applications", "~/.local/share/applications"};
-
-//
-static std::string resolvePath(std::string p) {
+static std::filesystem::path resolvePath(const std::string& p) {
     if (p[0] != '~')
         return p;
 
@@ -92,8 +89,10 @@ static std::string resolvePath(std::string p) {
     if (!HOME)
         return "";
 
-    return HOME + p.substr(1);
+    return std::filesystem::path(HOME) / p.substr(2);
 }
+
+static const std::array<std::filesystem::path, 3> DESKTOP_ENTRY_PATHS = {"/usr/local/share/applications", "/usr/share/applications", resolvePath("~/.local/share/applications")};
 
 CDesktopFinder::CDesktopFinder() : m_inotifyFd(inotify_init()), m_entryFrequencyCache(makeUnique<CEntryCache>("desktop")) {
     const auto ENV = getenv("XDG_DATA_DIRS");
@@ -103,8 +102,8 @@ CDesktopFinder::CDesktopFinder() : m_inotifyFd(inotify_init()), m_entryFrequency
     CConstVarList paths(ENV, 0, ':', false);
 
     for (const auto& p : paths) {
-        const auto      PTH = std::string{p} + "/applications";
-        std::error_code ec;
+        const std::filesystem::path PTH = std::filesystem::path(p) / "applications";
+        std::error_code             ec;
         if (!std::filesystem::exists(PTH, ec) || ec)
             continue;
 

